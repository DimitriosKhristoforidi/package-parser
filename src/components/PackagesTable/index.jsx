import { useContext, useEffect, useState } from 'react';
import { ContextApp } from '../../store/reducer';
import API from '../../API';
import { Table, Tag } from 'antd';
import { setLoading } from '../../store/actions';

const columns = [
  {
    title: 'Title',
    dataIndex: 'name',
    key: 'name',
    width: '15%',
  },
  {
    title: 'Description',
    dataIndex: 'description',
    key: 'description',
    width: '40%',
  },
  {
    title: 'Tags',
    dataIndex: 'keywords',
    key: 'keywords',
    render: keywords => (
      <>
          {
            keywords?.map(tag =>
              <Tag color="blue" key={tag}>
                {tag.toUpperCase()}
              </Tag>,
            )
          }
        </>
    ),
  },
  {
    title: 'NPM link',
    dataIndex: 'links',
    key: 'links',
    render: links => (
      <a target="_blank" rel="noreferrer" href={links.npm}>{links.npm}</a>
    ),
  },
];

const getPackagesData = async (state, dispatch, setPackagesData) => {
  if (state.packageList.length) {
    dispatch(setLoading(true));
    const promises = [];
    for (let i in state.packageList)
      promises.push(API.getPackage(state.packageList[i]));
    await Promise.all(promises).then(r => {
      const packagesData = r.map(item => item.results[0].package);
      setPackagesData(packagesData);
    });
    dispatch(setLoading(false));
  }
};

export default function PackagesTable() {
  const { dispatch, state } = useContext(ContextApp);
  const [packagesData, setPackagesData] = useState([]);

  useEffect(() => {
    getPackagesData(state, dispatch, setPackagesData);
  }, [state.packageList]);

  return (
    <div className="App">
      <Table
        loading={state.loading}
        columns={columns}
        dataSource={packagesData}
        pagination={false}
        bordered
        rowKey={'name'}
      />
    </div>
  );
}
