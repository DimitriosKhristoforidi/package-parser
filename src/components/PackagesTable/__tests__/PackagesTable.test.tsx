import React from 'react';
import { screen } from '@testing-library/react';
import PackagesTable from '../PackagesTable';
import { beforeEachTest } from '../../../utils/testsUtils';
import { render } from '../../../utils/testRender';
import { packagesListMock } from '../../../mock/testMocks';

describe('Packages Table', () => {
  beforeEach(beforeEachTest);

  const renderPackagesTable = () =>
    render(<PackagesTable />, {
      initialState: {
        packages: {
          loading: false,
          packagesList: packagesListMock,
        },
      },
    });

  it('Should: Render table headers', () => {
    renderPackagesTable();
    const tableHeading = screen.getByText('Title');
    expect(tableHeading.tagName.toLowerCase()).toBe('th');
  });

  it('Should: Render package line', () => {
    renderPackagesTable();
    const packageName = screen.getByText('react');
    expect(packageName.tagName.toLowerCase()).toBe('td');
  });
});
