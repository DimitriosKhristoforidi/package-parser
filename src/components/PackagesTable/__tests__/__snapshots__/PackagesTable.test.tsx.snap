// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Packages Table Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-rqglhn-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Title
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Description
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Tags
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                scope="col"
              >
                NPM link
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                scope="row"
              >
                react
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                React is a JavaScript library for building user interfaces.
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                <a
                  href="https://www.npmjs.com/package/react"
                  target="_blank"
                >
                  https://www.npmjs.com/package/react
                </a>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                scope="row"
              >
                react
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                React is a JavaScript library for building user interfaces.React is a JavaScript library for building user interfaces.React is a JavaScript library for building user interfaces.React is a JavaScript library for building user interfaces.React is a JavaScript library for building user interfaces.
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                <a
                  href="https://www.npmjs.com/package/react"
                  target="_blank"
                >
                  https://www.npmjs.com/package/react
                </a>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                scope="row"
              >
                react_react_react_react_react_react
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                React is a JavaScript library for building user interfaces.
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                    >
                      <span
                        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                      >
                        react
                      </span>
                    </div>
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
              >
                <a
                  href="https://www.npmjs.com/package/react"
                  target="_blank"
                >
                  https://www.npmjs.com/package/react
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root css-rqglhn-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Title
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Description
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Tags
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
              scope="col"
            >
              NPM link
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              scope="row"
            >
              react
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              React is a JavaScript library for building user interfaces.
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
            >
              <a
                href="https://www.npmjs.com/package/react"
                target="_blank"
              >
                https://www.npmjs.com/package/react
              </a>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              scope="row"
            >
              react
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              React is a JavaScript library for building user interfaces.React is a JavaScript library for building user interfaces.React is a JavaScript library for building user interfaces.React is a JavaScript library for building user interfaces.React is a JavaScript library for building user interfaces.
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
            >
              <a
                href="https://www.npmjs.com/package/react"
                target="_blank"
              >
                https://www.npmjs.com/package/react
              </a>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              scope="row"
            >
              react_react_react_react_react_react
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              React is a JavaScript library for building user interfaces.
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      react
                    </span>
                  </div>
                </div>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
            >
              <a
                href="https://www.npmjs.com/package/react"
                target="_blank"
              >
                https://www.npmjs.com/package/react
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
